### Exercício 6 
 
 #### Teste a hipótese de o salário ser o mesmo independente do grau de instrução e estado civil. Qual sua conclusão?

In [1]:
# importa biblioteca pandas
import pandas as pd

# importa biblioteca statsmodels
import statsmodels.api as sm

# local do arquivo
Caminho='/Users/fabicampanari/Desktop/_8-Prova Matematematica/1-statiscalMeasures_ Hypothesis Testing II./1🇧🇷-statiscalMeasures_ Hypothesis Testing II./answeredCodes_statiscalMeasures/cadastro_funcionarios.xlsx'


In [2]:
#Traz o arquivo para o Python
df=pd.read_excel(Caminho)


In [3]:
df.head( )


ID estado_civil grau_instrucao  n_filhos  salario  idade reg_proc
0   1            s              f         0     4.00     26      int
1   2            c              f         1     4.56     32      cap
2   3            c              f         2     5.25     36      cap
3   4            s              m         0     5.73     20      out
4   5            s              f         0     6.26     40      out

In [4]:
#Verifica a média de idade por grau de instrução
df.groupby(['grau_instrucao'])['idade'].describe()




count       mean       std   min   25%   50%   75%   max
grau_instrucao                                                          
f               120.0  35.533333  6.925696  20.0  28.0  37.0  41.0  49.0
m               203.0  34.556650  6.145955  20.0  31.0  35.0  37.0  49.0
s               103.0  33.689320  6.159687  22.0  31.0  33.0  38.0  49.0

In [5]:
#Verifica a média de idade por estado_civil
df.groupby(['estado_civil'])['idade'].describe()

count       mean       std   min   25%   50%   75%   max
estado_civil                                                          
c             244.0  35.094262  5.761670  25.0  31.0  35.0  39.0  49.0
s             182.0  33.989011  7.130968  20.0  27.0  34.0  41.0  46.0

In [6]:
#Gera o modelo para comparar idade por grau de instrução e região
from statsmodels.formula.api import ols
modelo=ols('idade ~ grau_instrucao + estado_civil ',data=df).fit()


In [7]:
#Aplicação ANOVA Testando H0: μf= μm=μs   H1: Ao menos um diferente
#H0: μc= μs   H1: Ao menos um diferente

anova=sm.stats.anova_lm(modelo)
print(anova)


                   df        sum_sq     mean_sq         F    PR(>F)
grau_instrucao    2.0    190.129141   95.064570  2.351952  0.096430
estado_civil      1.0    151.018755  151.018755  3.736290  0.053911
Residual        422.0  17057.004687   40.419442       NaN       NaN


In [8]:
#Teste Post Hoc de Tukey para avaliar diferenças no estado civil
from statsmodels.stats.multicomp import pairwise_tukeyhsd
tukey=pairwise_tukeyhsd(endog=df.idade, groups=df.estado_civil)
tukey.summary ( )


<class 'statsmodels.iolib.table.SimpleTable'>

In [9]:

#Teste Post Hoc de Tukey para avaliar diferenças no grau de instrução
from statsmodels.stats.multicomp import pairwise_tukeyhsd
tukey=pairwise_tukeyhsd(endog=df.idade, groups=df.grau_instrucao)
tukey.summary ( )



<class 'statsmodels.iolib.table.SimpleTable'>

In [10]:
#Gera o modelo para comparar idade por grau de instrução e região
from statsmodels.formula.api import ols
modelo=ols('salario ~ grau_instrucao + estado_civil ',data=df).fit()


In [11]:
anova=sm.stats.anova_lm(modelo)
print(anova)


                   df       sum_sq      mean_sq           F        PR(>F)
grau_instrucao    2.0  2409.879383  1204.939691  103.923621  2.005826e-37
estado_civil      1.0   251.442334   251.442334   21.686395  4.308139e-06
Residual        422.0  4892.867899    11.594474         NaN           NaN


In [12]:
#Teste Post Hoc de Tukey para avaliar diferenças no estado civil
from statsmodels.stats.multicomp import pairwise_tukeyhsd
tukey=pairwise_tukeyhsd(endog=df.salario, groups=df.estado_civil)
tukey.summary ( )


<class 'statsmodels.iolib.table.SimpleTable'>

In [13]:
#Teste Post Hoc de Tukey para avaliar diferenças no grau de instrução
from statsmodels.stats.multicomp import pairwise_tukeyhsd
tukey=pairwise_tukeyhsd(endog=df.salario, groups=df.grau_instrucao)
tukey.summary ( )


<class 'statsmodels.iolib.table.SimpleTable'>

## Resposta Qustao 6

### Passo 1: Importar Bibliotecas Necessárias

In [14]:
# Importar bibliotecas necessárias
import pandas as pd
import statsmodels.api as sm
from statsmodels.formula.api import ols
from statsmodels.stats.multicomp import pairwise_tukeyhsd

#### Passo 2: Carregar os Dados do Arquivo Excel


In [15]:
# Local do arquivo
Caminho = '/Users/fabicampanari/Desktop/_8-Prova Matematematica/1-statiscalMeasures_ Hypothesis Testing II./1🇧🇷-statiscalMeasures_ Hypothesis Testing II./answeredCodes_statiscalMeasures/cadastro_funcionarios.xlsx'

# Carregar o arquivo para o DataFrame
df = pd.read_excel(Caminho)

#### Passo 3: Visualizar as Primeiras Linhas do DataFrame

In [16]:
# Visualizar as primeiras linhas do DataFrame
print(df.head())

   ID estado_civil grau_instrucao  n_filhos  salario  idade reg_proc
0   1            s              f         0     4.00     26      int
1   2            c              f         1     4.56     32      cap
2   3            c              f         2     5.25     36      cap
3   4            s              m         0     5.73     20      out
4   5            s              f         0     6.26     40      out


#### Passo 4: Verificar a Média de Salário por Grau de Instrução e Estado Civil

In [17]:
# Verificar a média de salário por grau de instrução e estado civil
print(df.groupby(['grau_instrucao', 'estado_civil'])['salario'].describe())

                             count       mean       std   min     25%     50%  \
grau_instrucao estado_civil                                                     
f              c              57.0   8.048596  2.918592  4.00   5.450   7.600   
               s              63.0   7.808730  2.712180  4.00   6.260   7.200   
m              c             131.0  12.223664  4.014562  4.43   8.925  12.790   
               s              72.0   9.361528  2.428721  5.73   7.440   9.075   
s              c              56.0  14.863929  3.450633  9.50  12.790  14.225   
               s              47.0  14.083617  3.743024  7.60  10.565  15.320   

                               75%    max  
grau_instrucao estado_civil                
f              c              9.40  14.70  
               s              8.48  13.90  
m              c             15.20  23.90  
               s             10.76  16.07  
s              c             17.26  23.50  
               s             17.25  18.90  

#### Passo 5: Gerar o Modelo para Comparar Salário por Grau de Instrução e Estado Civil

In [18]:
# Gerar o modelo para comparar salário por grau de instrução e estado civil
modelo = ols('salario ~ grau_instrucao + estado_civil', data=df).fit()
modelo

#### Passo 6: Aplicar ANOVA para Testar a Hipótese

In [19]:
# Aplicar ANOVA
anova = sm.stats.anova_lm(modelo)
print("Resultados do Teste ANOVA:")
print(anova)

Resultados do Teste ANOVA:
                   df       sum_sq      mean_sq           F        PR(>F)
grau_instrucao    2.0  2409.879383  1204.939691  103.923621  2.005826e-37
estado_civil      1.0   251.442334   251.442334   21.686395  4.308139e-06
Residual        422.0  4892.867899    11.594474         NaN           NaN


#### Passo 7: Realizar o Teste Post Hoc de Tukey para Avaliar Diferenças Específicas

In [20]:
# Teste Post Hoc de Tukey para avaliar diferenças no estado civil
print("\nResultados do Teste Post Hoc de Tukey para Estado Civil:")
tukey_estado_civil = pairwise_tukeyhsd(endog=df.salario, groups=df.estado_civil)
print(tukey_estado_civil.summary())


Resultados do Teste Post Hoc de Tukey para Estado Civil:
Multiple Comparison of Means - Tukey HSD, FWER=0.05
group1 group2 meandiff p-adj  lower   upper  reject
---------------------------------------------------
     c      s  -1.8108   0.0 -2.6048 -1.0168   True
---------------------------------------------------


In [21]:
# Teste Post Hoc de Tukey para avaliar diferenças no grau de instrução
print("\nResultados do Teste Post Hoc de Tukey para Grau de Instrução:")
tukey_instrucao = pairwise_tukeyhsd(endog=df.salario, groups=df.grau_instrucao)
print(tukey_instrucao.summary())


Resultados do Teste Post Hoc de Tukey para Grau de Instrução:
Multiple Comparison of Means - Tukey HSD, FWER=0.05
group1 group2 meandiff p-adj lower  upper  reject
-------------------------------------------------
     f      m   3.2859   0.0 2.3414 4.2303   True
     f      s   6.5852   0.0 5.4835 7.6869   True
     m      s   3.2993   0.0 2.3071 4.2916   True
-------------------------------------------------


Conclusão

### Resultados do Teste ANOVA:

| Source         | df  | sum_sq      | mean_sq     | F          | PR(>F)         |
|----------------|-----|-------------|-------------|------------|----------------|
| grau_instrucao | 2.0 | 2409.879383 | 1204.939691 | 103.923621 | 2.005826e-37   |
| estado_civil   | 1.0 | 251.442334  | 251.442334  | 21.686395  | 4.308139e-06   |
| Residual       | 422 | 4892.867899 | 11.594474   | NaN        | NaN            |

#### Interpretação:

**Para Grau de Instrução:**

- Valor F: 103.923621
- Valor p (PR(>F)): 2.005826e-37

**Para Estado Civil:**

- Valor F: 21.686395
- Valor p (PR(>F)): 4.308139e-06

Os valores p são extremamente pequenos, muito menores que o nível de significância comum (0.05). Isso indica que há uma diferença significativa nos salários tanto para diferentes graus de instrução quanto para diferentes estados civis.


### Resultados do Teste Post Hoc de Tukey para Grau de Instrução:
##### Multiple Comparison of Means - Tukey HSD, FWER=0.05
| G1 | G2 | Mean Diff | p-adj | Lower | Upper | Reject |
|----|----|-----------|-------|-------|-------|--------|
| f  | m  | 3.29      | 0.0   | 2.34  | 4.23  | True   |
| f  | s  | 6.59      | 0.0   | 5.48  | 7.69  | True   |
| m  | s  | 3.30      | 0.0   | 2.31  | 4.29  | True   |

<br><br>

### Resultados do Teste Post Hoc de Tukey para Estado Civil:

##### Multiple Comparison of Means - Tukey HSD, FWER=0.05
| group1 | group2 | meandiff | p-adj | lower  | upper  | reject |
|--------|--------|----------|-------|--------|--------|--------|
| c      | s      | -1.8108  | 0.0   | -2.6048| -1.0168| True   |



<br><br>

Conclusão:

Os resultados do teste ANOVA indicam que há diferenças significativas nos salários tanto para diferentes graus de instrução quanto para diferentes estados civis. Os testes Post Hoc de Tukey confirmam que:

Para grau de instrução, todas as comparações entre os grupos (f, m, s) são estatisticamente significativas.
Para estado civil, a comparação entre casados (c) e solteiros (s) é estatisticamente significativa.
Portanto, a conclusão é que o salário não é o mesmo independentemente do grau de instrução e estado civil. Existem diferenças significativas nos salários com base nesses fatores.